drake_ros_tf2/src/utilities/name_conventions.cc

```diff
@@ -0,0 +1,79 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "drake_ros_tf2/utilities/name_conventions.h"
+
+#include <sstream>
+#include <string>
+#include <unordered_set>
+
+namespace drake_ros_tf2 {
+namespace utilities {
+
+namespace {
+
+std::string ReplaceAllOccurrences(std::string string, const std::string& target,
+                                  const std::string& replacement) {
+  std::string::size_type n = 0;
+  while ((n = string.find(target, n)) != std::string::npos) {
+    string.replace(n, target.size(), replacement);
+    n += replacement.size();
+  }
+  return string;
+}
+
+}  // namespace
+
+std::string GetTfFrameName(
+    const drake::geometry::SceneGraphInspector<double>& inspector,
+    const std::unordered_set<const drake::multibody::MultibodyPlant<double>*>&
+        plants,
+    const drake::geometry::FrameId& frame_id) {
+  std::stringstream ss;
+  for (auto* plant : plants) {
```

Comment:
> It seems like the same frame name will be generated If multiple multibody plants have the same model. How big of a problem is that?

You're right. Just to make sure I understand the problem, model/body instance names do not need to be unique across all multibody plants within the same `SceneGraph`, is that correct?

> What do you think of putting the frame_id into the string frame name?

Sounds like this might be a good solution to get us overall unique frame names. I'll take a crack at it.

Reply:
Appending the value of the frame ID after the body name, https://github.com/RobotLocomotion/drake-ros/pull/35/commits/272cbe56a28e6ece84f14517d4a27be7f318f2ab